velero-plugin-for-microsoft-azure/object_store.go

```diff
@@ -345,7 +362,7 @@ func (o *ObjectStore) Init(config map[string]string) error {
 	}
 
 	o.log.Debugf("Creating service client")
-	serviceClient, o.sharedKeyCredential, err = getServiceClient(o.storageAccount, config[subscriptionIDConfigKey], config[resourceGroupConfigKey], o.sharedKeyCredential, cloudConfig)
+	serviceClient, o.sharedKeyCredential, err = getServiceClient(o.storageAccount, config[subscriptionIDConfigKey], config[resourceGroupConfigKey], config[storageAccountURIConfigKey], config[useAADConfigKey], o.sharedKeyCredential, cloudConfig, o.log)
```

Comment:
As the `subscriptionId` is optional, we should read it from the env if it isn't set in the BSL. 

There is a bug in the code of 1.7 which only reads it from the BSL, I'll fix it in 1.7 branch later.

Refer to https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/blob/release-1.6/velero-plugin-for-microsoft-azure/object_store.go#L190

Reply:
Got it
SubscriptionID will fallback to environment if not provided in BSL

Is there anything I need to address as part of this current PR?

The commands should still retain the subID field right, because it is possible for user to be choose a Storage Account in a different Subscription. IMO sample commands hence should still show how to set it.

Reply:
Yes, I'll raise the fix for subscription fallback as a separate PR.

Reply:
Thanks, @anshulahuja98 , let me handle this issue. I have submitted a fix for branch 1.11 and will fix it in the main branch after this PR is merged in the same way